spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeletesRewrite.java

```diff
@@ -362,10 +367,18 @@ public void write(InternalRow record) throws IOException {
       InternalRow row = record.getStruct(rowOrdinal, rowSize);
       if (row != null) {
         positionDelete.set(file, position, row);
-        lazyWriterWithRow().write(positionDelete, spec, partition);
+        if (formatVersion >= 3) {
```

Comment:
instead of checking the format version here we can alternatively have a separate `private static class DVWriter implements DataWriter<InternalRow>` (similar to the current `DeleteWriter`). We'd then have the following code in L246ff:


```
      if (TableUtil.formatVersion(table) >= 3) {
        return new DVWriter(table, deleteFileFactory, dsSchema, specId, partition);
      }

      return new DeleteWriter(
          table,
          writerFactoryWithRow,
          // other params
```